src/libpriv/rpmostree-postprocess.c

```diff
@@ -512,34 +512,68 @@ rpmostree_prepare_rootfs_get_sepolicy (int            dfd,
   return ret;
 }
 
-static gboolean
-replace_nsswitch (int            dfd,
-                  GCancellable  *cancellable,
-                  GError       **error)
+char *
+rpmostree_postprocess_replace_nsswitch_string (const char *buf,
+                                               GError    **error)
 {
-  g_autofree char *nsswitch_contents = NULL;
-  g_autofree char *new_nsswitch_contents = NULL;
+  gboolean is_passwd;
+  gboolean is_group;
+
+  is_passwd = g_str_has_prefix (buf, "passwd:");
+  is_group = g_str_has_prefix (buf, "group:");
```

Comment:
Maybe the easiest thing to do here is to still use `g_regex` to get the passwd/group lines to feed to this function, and then `g_regex_replace` them?